### DISCUSSION TITLE
Why is ~/.vim/vimrc read before ./defaults.vim?

### DISCUSSION THREAD

**Discussion by git-madresc** [vim/vim#15001] [Q&A]
For me, this question is in no way new, vim does this as long as I remember. However, some distributions have defaults defined, that I do not want. like 'set mouse=a'. And I can not disable this behavior in my ~/.vim/vimrc.

So, my questions would be:
- Is there any good reason, why ~/.vim/vimrc is not processed last?
- Is there any way, this order could change at some point in the future?

**Comment by chrisbra** [comment]
If you have a custom vimrc, defaults.vim should not be read. If it still is, this must be configured in your environment specifically.

**Reply by git-madresc** [reply]
Oh, this was quick and simple, thank you!
Has this always been the case? Because, I feel (yes, I know...), some years ago at some version 7, this was driving me crazy.

**Reply by chrisbra** [reply]
defaults.vim is relatively new, perhaps around 8.something?

**Reply by git-madresc** [reply]
I answer with a qoute: "It's (time) like a predator; it's stalking you."
Looking at the sources, it has been the for at least 8 years, some version 7.4.

But thank you anyway, I had no intention to waste your time. Frohes Schaffen, schönen Feierabend!
